src/components/ModalBase.component.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { ReactNode } from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  StyleProp,
+  ViewStyle
+} from 'react-native';
+import Modal from 'react-native-modal';
+
+import { COLORS } from '../styles/themes';
+import NewHathorButton from './NewHathorButton';
+
+const ModalBase = ({ styleModal, styleWrapper, show, onDismiss, children }) => {
+  const hasChildren = children != null;
+
+  const title = hasChildren && React.Children.toArray(children).find(
+    (child) => child.type.displayName === Title.displayName
+  );
+  const body = hasChildren && React.Children.toArray(children).find(
+    (child) => child.type.displayName === Body.displayName
+  );
+  const button = hasChildren && React.Children.toArray(children).find(
+    (child) => child.type.displayName === Button.displayName
+  );
+  const discreteButton = hasChildren && React.Children.toArray(children).find(
+    (child) => child.type.displayName === DiscreteButton.displayName
+  );
+
+  return (
+    <Modal
+      isVisible={show}
+      animationIn='slideInUp'
+      swipeDirection={['down']}
+      onSwipeComplete={onDismiss}
+      onBackButtonPress={onDismiss}
+      onBackdropPress={onDismiss}
+      style={styleModal}
+      propagateSwipe
+    >
+      <View style={[
+        styles.wrapper,
+        styleWrapper,
+      ]}
+      >
+        {title && title}
+        {body && body}
+        {button && button}
+        {discreteButton && discreteButton}
+      </View>
+    </Modal>
+  );
+};
+
+const Title = ({ children }) => (
+  <View style={styles.titleWrapper}>
+    <Text style={styles.title}>
+      {children}
+    </Text>
+  </View>
+);
+Title.displayName = 'ModalBaseTitle';
+
+/**
+ * @typedef {Object} Properties p
+ * @property {ReactNode} p.children
+ * @property {StyleProp<ViewStyle>} p.style
+ *
+ * @param {Properties}
```

Comment:
Where is Properties defined? Is it from react-native? LSPs understands this?

Reply:
Fixed.